burn-tensor/src/tensor/api/base.rs

```diff
@@ -212,6 +307,19 @@ pub trait BasicOps<B: Backend>: TensorKind<B> {
         rhs: Self::Primitive<D>,
     ) -> Tensor<B, D, Bool>;
     fn equal_elem<const D: usize>(lhs: Self::Primitive<D>, rhs: Self::Elem) -> Tensor<B, D, Bool>;
+    fn elem_type() -> &'static str {
+        if TypeId::of::<Self::Elem>() == TypeId::of::<f32>() {
```

Comment:
@nathanielsimard Also here I assume I'm missing a bunch of other possibilities like f64 for float, or i8, i16, i128, etc. for ints. What else should I add?

Reply:
You can use directly `core::any::type_name::<Seld::Elem>()` and all types will be supported :)

Reply:
Do you want the more specific type that comes from `core::any::type_name::<Self::Elem>()`, or should I group the possible returns as int or float?

Reply:
The more specific type is needed, but we could also add the tensor kind:

```
  data: [...],
  kind: Float, // We could add a function `name` in the trait TensorKind.
  elem: f32
```

Reply:
Sounds good, just threw kind in as well.